python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -649,6 +649,19 @@ def test_timestamps_with_timezone(self):
 
         _check_pandas_roundtrip(df)
 
+    def test_python_datetime(self):
+        # ARROW-2106
+        date_array = [datetime.today() + timedelta(days=x) for x in range(10)]
+        df = pd.DataFrame({
+            'datetime': pd.Series(date_array, dtype=object)
+        })
+        result = pa.Table.from_pandas(df).to_pandas()
```

Comment:
Can you add an assertion that the column in the result of `pyarrow.Table.from_pandas` is `TimestampArray` (per the comment above). I expect that is already the case